Makefile

```diff
@@ -3,7 +3,8 @@
 #
 
 os_name := $(shell uname -s)
-rust_intended := 1.52.0
+rustup_exists := $(shell which rustup)
+rust_intended := 1.52.1
```

Comment:
Not sure how you committed this... I would think that `make version_check` would fail due to the wrong rust version in the GitHub action files. And, the `version_check` should be run as part of `make precommit`.